rmw_fastrtps_cpp/src/type_support_common.cpp

```diff
@@ -119,6 +119,11 @@ bool TypeSupport::deserializeROSmessage(
       "Fast CDR exception deserializing message of type %s.",
       getName());
     return false;
+  } catch (...) {
```

Comment:
Generally, we would prefer to have specific exceptions listed here, rather than just catch everything.

What exceptions can this deserialization throw?

Reply:
Looking at the original report, it looks like we should just catch `std::bad_alloc` for now.  @MiguelCompany if you make that change and rebase this PR onto the latest, I'll be happy to approve and run CI on it.